wgsl/index.bs

```diff
@@ -9107,20 +9119,54 @@ the implementation can issue a clear diagnostic.
   </xmp>
 </div>
 
-## Extensions List ## {#extension-list}
+## Software Extensions ## {#software-extensions-sec}
+
+A <dfn noexport>software extension</dfn> is an [=extension=] which is automatically available if the implementation supports it.
+The program does not have to explicitly request it.
+
+[=Software extensions=] embody functionality which could reasonably be supported on any WebGPU implementation.
+If the feature is not universally available, that it is because some WebGPU implementation has not yet implemented it.
+
+TODO: https://github.com/gpuweb/gpuweb/issues/3875 How does WebGPU signal support for a software extension?
+
+A <dfn noexport>requires-directive</dfn> is a [=directive=] that *documents* the program's use of one or more [=software extensions=].
+It does not change the functionality exposed by the implementation.
+A [=shader-creation error=] results if the implementation does not support one of the extensions.
+
+A WGSL program *can* use a [=requires-directive=] to signal the potential for non-portability,
+and to signal the *intended* minimum bar for portability.
+
+Note: Tooling outside of a WebGPU implementation could check whether all the [=software extensions=] used by a
+program are covered by [=requires-directives=] in the program.
+
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>requires_directive</dfn> :
+
+    | <a for=syntax_kw lt=enable>`'requires'`</a> [=syntax/software_extension_list=] <a for=syntax_sym lt=semicolon>`';'`</a>
```

Comment:
From a symmetry perspective, why is this a list, but enable extensions are not?

Reply:
Generalized it to expand to ident_pattern_token.